### DISCUSSION TITLE
Raven Solvers

### DISCUSSION THREAD

**Discussion by brunorbqi** [SysBioChalmers/RAVEN#486] [Help me!] [answered]
Hello, I am installing the new version of RAVEN Toolbox, 2.8.0 and I am having the following problem, all model solvers are fail. Even the glpk, andI have a valid gurobi license. How can I fix this ? Best regards.
![Captura de tela de 2023-05-17 07-29-50](https://github.com/SysBioChalmers/RAVEN/assets/68126532/dd9e0c05-8067-4310-ba64-1cd418ce5b7c)

**Comment by edkerk** [comment]
Could you please report the output from `runtests('solverTests')` ?

**Reply by brunorbqi** [reply]
Running solverTests

================================================================================
Error occurred in solverTests/testGlpk and it did not run to completion.

    ---------
    Error ID:
    ---------
    ''

    --------------
    Error Details:
    --------------
    Error using solverTests>testGlpk (line 27)
    Solver not working
================================================================================
.
================================================================================
Error occurred in solverTests/testGurobi and it did not run to completion.

    ---------
    Error ID:
    ---------
    ''

    --------------
    Error Details:
    --------------
    Error using solverTests>testGurobi (line 63)
    Solver not working
================================================================================
.
================================================================================
Error occurred in solverTests/testSoplex and it did not run to completion.

    ---------
    Error ID:
    ---------
    ''

    --------------
    Error Details:
    --------------
    Error using solverTests>testSoplex (line 79)
    SoPlex not installed or cannot be found, test skipped
================================================================================
.
================================================================================
Error occurred in solverTests/testCobra and it did not run to completion.

    ---------
    Error ID:
    ---------
    ''

    --------------
    Error Details:
    --------------
    Error using solverTests>testCobra (line 140)
    Solver not working
================================================================================
.
Done solverTests
__________

Failure Summary:

     Name                    Failed  Incomplete  Reason(s)
    =======================================================
     solverTests/testGlpk      X         X       Errored.
    -------------------------------------------------------
     solverTests/testGurobi    X         X       Errored.
    -------------------------------------------------------
     solverTests/testSoplex    X         X       Errored.
    -------------------------------------------------------
     solverTests/testCobra     X         X       Errored.

ans = 

  1×4 TestResult array with properties:

    Name
    Passed
    Failed
    Incomplete
    Duration
    Details

Totals:
   0 Passed, 4 Failed (rerun), 4 Incomplete.
   0.8659 seconds testing time.

**Reply by edkerk** [reply]
Ok, so that didn't give more useful information. Could you instead please try the following?
```
load(fullfile(findRAVENroot,'testing','unit_tests','test_data','ecoli_textbook.mat'));
setRavenSolver('glpk')
solveLP(model)
```

**Reply by brunorbqi** [reply]
>> load(fullfile(findRAVENroot, 'testing','unit_tests','test_data','ecoli_textbook.mat'));
>> setRavenSolver('glpk')
>> solverLP(model)
Undefined function or variable 'solverLP'.
 
Did you mean:
>> solveLP(model)
Error: File: optimizeProb.m Line: 195 Column: 17
"i" previously appeared to be used as a function or command, conflicting with its use here as the name of a
variable.
A possible cause of this error is that you forgot to initialize the variable, or you have initialized it implicitly
using load or eval.


Error in solveLP (line 103)
res = optimizeProb(prob,params);

**Reply by edkerk** [reply]
The error message reported in your previous comment should now be fixed in [RAVEN 2.8.1](https://github.com/SysBioChalmers/RAVEN/releases/tag/v2.8.1).

However, your reported error occurs when using 'soplex' as solver. Could you please double-check and confirm what error message you get if you first run `setRavenSolver('glpk')` or `setRavenSolver('gurobi')`  ?

**Reply by edkerk** [reply, marked as answer]
Actually, you report the same behaviour as in https://github.com/SysBioChalmers/RAVEN/issues/478. There also all solvers were reported to fail. This was resolved by the fix that is included in RAVEN 2.8.1, so this will most likely also resolve the issue for you.

**Reply by brunorbqi** [reply]
I updated the RAVEN to the latest version and worked very fine. Thank you very much for your atention !!! Best regards.

**Comment by gaeaberry** [comment]
Hello.
I am new in RAVEN toolbox. I am getting this error. I already downloaded gurobi optimizer too. But I cannot do the tutorial of RAVEN toolbox.
Can you help me with this?
Thanks.
![Adsız](https://github.com/user-attachments/assets/0070232b-0e9c-42ec-985c-b4c23b9181f6)
